src/App.tsx

```diff
@@ -1,61 +1,133 @@
 import './App.scss';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { TodoList } from './components/TodoList';
+
+function getUserById(userId: number): User | null {
+  const correctUser = usersFromServer.find(user => user.id === userId);
+
+  return correctUser || null;
+}
+
+function getUserByName(userName: string) {
+  const correctUser = usersFromServer.find(({ name }) => userName === name);
+
+  return correctUser || null;
+}
+
+export const preparedTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [todos, setTodos] = useState<Todo[]>(preparedTodos);
+  const [selectedUser, setSelectedUser] = useState('');
+  const [errorMessage, setErrorMessage] = useState(false);
+
+  const addTodo = () => {
+    const correctUser = getUserByName(selectedUser);
+
+    if (correctUser) {
+      const newId = Math.max(...todos.map(todo => todo.id)) + 1;
+
+      const newTodo: Todo = {
+        id: newId,
+        userId: correctUser.id,
+        title,
+        completed: false,
+        user: correctUser,
+      };
+
+      setTodos((prevTodos) => [...prevTodos, newTodo]);
+    }
+  };
+
+  const clearForm = () => {
+    setSelectedUser('');
+    setTitle('');
+    setErrorMessage(false);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUser(event.target.value);
+  };
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title || !selectedUser) {
+      setErrorMessage(true);
+
+      return;
+    }
+
+    addTodo();
+    clearForm();
+  };
+
+  const titleErrorStatus = !title && errorMessage;
```

Comment:
will be better to name these variables as a boolean ones
for now as user I expect that such variables can contain some status in `string` format, not in the boolean
you can use `hasTitleError`, for example